ci/ray_ci/automation/ray_wheels_lib.py

```diff
@@ -0,0 +1,89 @@
+import boto3
+from typing import List
+import os
+
+from ci.ray_ci.utils import logger
+
+bazel_workspace_dir = os.environ.get("BUILD_WORKSPACE_DIRECTORY", "")
+
+PYTHON_VERSIONS = ["cp38-cp38", "cp39-cp39", "cp310-cp310", "cp311-cp311"]
+PLATFORMS = [
+    "manylinux2014_x86_64",
+    "manylinux2014_aarch64",
+    "macosx_10_15_x86_64",
+    "macosx_11_0_arm64",
+    "win_amd64",
+]
+RAY_TYPES = ["ray", "ray_cpp"]
+
+
+def _check_downloaded_wheels(directory_path: str, wheels: List[str]) -> None:
+    """Check if the wheels are downloaded properly."""
+    assert os.path.exists(directory_path), f"{directory_path} does not exist"
+
+    for wheel in wheels:
+        wheel_path = os.path.join(directory_path, wheel + ".whl")
+        if not os.path.exists(wheel_path):
```

Comment:
if's often better to return a boolean and the error message instead of throwing errors to the caller (which can also happen because of any other reasons)

Reply:
fixed! I also return boolean & message for all functions in this file as well, to pass errors along them better.

Reply:
can we discuss? like since when we started doing bool+string thing? I have never seen you using this pattern before @can-anyscale ..

Reply:
ping @can-anyscale  did we end up making a decision on which pattern to use?